Add changes from reviewing the defense section (#35)
xsleaks/wiki: content/docs/defenses/opt-in/coop.md

menu = "main"
+++

> Getting access to a website's `window` object is a common prerequisite for different XS-Leak techniques. [Framing Protections]({{< ref "xfo.md" >}}) can ensure that an attacker cannot use iframes to access the `window` object, but this does not stop an attacker from using `window.open` or `window.opener`. 

Author: Tried to simplify this and get to the point a bit quicker

> Exploiting XS-Leaks with `window.open` is generally seen as the least appealing option because the user can see it happen in the open browser window. However, it's usually the right technique when:

Reviewer: ```suggestion
Exploiting XS-Leaks with `window.open` is generally seen as less appealing option because the user might notice it happen in the opened browser window. This technique however can be improved with combination of [Pop-unders](https://www.webopedia.com/TERM/P/popunder.html) or by utilising `window.opener` reference which will likely be processed in the background. Nevertheless, it's usually the right technique when:
```

- A page sets [Framing Protections]({{< ref "xfo.md" >}})

> - A page sets [Same-Site Cookies with `Lax` Mode]({{< ref "same-site-cookies.md" >}}) (since navigating a new window is a top-level navigation, it bypasses this)

Reviewer: ```suggestion
- A page sets [Same-Site Cookies with `Lax` Mode]({{< ref "same-site-cookies.md" >}}) (in contrast to the `Strict` mode, navigating a top-level window is allowed by the `Lax` one)
```

> To prevent other websites from gaining arbitrary window references to a page, applications can deploy Cross-Origin-Opener-Policy (COOP) [^1] [^2]. With COOP, applications are in control of who is allowed to have a reference to their pages, thus mitigating XS-Leaks which make use of such techniques.

Reviewer: ```suggestion
To prevent cross-origin websites from gaining arbitrary window references to a page, applications can deploy Cross-Origin-Opener-Policy (COOP) [^1] [^2] that is meant for that purpose, thus mitigating XS-Leaks which make use of such techniques.
```

I removed one sentence since it's quite confusing. The website is not fully in control of who can access the reference, it's very limited in fact. The only possible values are:
* same-origin (only allow same-origin pages)
* same-origin-allow-popups (same-origin + opened popups)
* unsafe-none (the protection is disabled)

I am surprised this isn't mentioned in the article and it seems as the most crucial information to be put in the section. 

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cross-Origin-Opener-Policy

Author: Will add more details in a follow up PR

## Considerations

> Since COOP is an opt-in mechanism and a very recent one, it can easily be overlooked by developers and security engineers. Nonetheless, it’s important to highlight the importance of this defense mechanism as it is the only way to prevent attackers from exploiting XS-Leaks which makes use of window references returned by APIs like `window.open` (unless Same-Site Cookies Strict Mode can be widely deployed).

Reviewer: ```suggestion
Since COOP is an opt-in mechanism and a very recent one, it can easily be overlooked by developers and security engineers. Nonetheless, it’s important to highlight the importance of this defense mechanism as it is currently the best possible way to prevent attackers from exploiting XS-Leaks utilising window references returned.
```
